src/mongo/db/backup/backup_commands.cpp

```diff
@@ -43,12 +44,13 @@ class CreateBackupCommand : public Command {
              << std::endl
              << "{ createBackup: 1, backupDir: <destination directory> }";
     }
-    void addRequiredPrivileges(const std::string& dbname,
-                               const BSONObj& cmdObj,
-                               std::vector<Privilege>* out) override {
-        ActionSet actions;
-        actions.addAction(ActionType::startBackup);
-        out->push_back(Privilege(parseResourcePattern(dbname, cmdObj), actions));
+    Status checkAuthForCommand(ClientBasic* client,
+                               const std::string& dbname,
+                               const BSONObj& cmdObj) override {
+        return AuthorizationSession::get(client)->isAuthorizedForActionsOnResource(
+                   ResourcePattern::forAnyNormalResource(), ActionType::startBackup)
+            ? Status::OK()
+            : Status(ErrorCodes::Unauthorized, "Unauthorized");
```

Comment:
Thank you for details.
Unfortunately I don't understand why previous code only checks against `admin` db. Admin db is not mentioned there. But I see that new code effectively checks if authorized user has `startBackup` action in his roles.
One interesting thing is that `dbname` parameter is not used. In my understanding this is because `startBackup` works on a higher level (data directory).